test/serializer-test.ts

```diff
@@ -50,6 +50,92 @@ describe("serializer", function(): void {
     assert.deepEqual(serialized, expectedJSON);
   });
 
+  it("serializes a payment to an X-address with a tag in XRP", function(): void {
+    // GIVEN a transaction which represents a payment to a destination and tag, denominated in XRP.
+    const value = "1000";
+    const destination = "XVfC9CTCJh6GN2x8bnrw3LtdbqiVCUvtU3HnooQDgBnUpQT";
+    const fee = "10";
+    const sequence = 1;
+    const account = "r9LqNeG6qHxjeUocjvVki2XR35weJ9mZgQ";
+    const publicKey = "testPublicKey";
+
+    const paymentAmount = new XRPAmount();
+    paymentAmount.setDrops(value);
+
+    const payment = new Payment();
+    payment.setDestination(destination);
+    payment.setXrpAmount(paymentAmount);
+
+    const transactionFee = new XRPAmount();
+    transactionFee.setDrops(fee);
+
+    const transaction = new Transaction();
+    transaction.setAccount(account);
+    transaction.setFee(transactionFee);
+    transaction.setSequence(sequence);
+    transaction.setPayment(payment);
+    transaction.setSigningPublicKeyHex(publicKey);
+
+    // WHEN the transaction is serialized to JSON.
+    const serialized = Serializer.transactionToJSON(transaction);
+
+    // THEN the result is as expected.
+    const expectedJSON = {
+      Account: account,
+      Amount: value + "",
+      Destination: "rU6K7V3Po4snVhBBaU29sesqs2qTQJWDw1",
+      DestinationTag: 12345,
+      Fee: fee,
+      Sequence: sequence,
+      TransactionType: "Payment",
+      SigningPubKey: publicKey
+    };
+    assert.deepEqual(serialized, expectedJSON);
+  });
+
+  it("serializes a payment to an X-address without a tag in XRP", function(): void {
+    // GIVEN a transaction which represents a payment to a destination without a tag, denominated in XRP.
+    const value = "1000";
+    const destination = "XVfC9CTCJh6GN2x8bnrw3LtdbqiVCUFyQVMzRrMGUZpokKH";
+    const fee = "10";
+    const sequence = 1;
+    const account = "r9LqNeG6qHxjeUocjvVki2XR35weJ9mZgQ";
+    const publicKey = "testPublicKey";
+
+    const paymentAmount = new XRPAmount();
+    paymentAmount.setDrops(value);
+
+    const payment = new Payment();
+    payment.setDestination(destination);
+    payment.setXrpAmount(paymentAmount);
+
+    const transactionFee = new XRPAmount();
+    transactionFee.setDrops(fee);
+
+    const transaction = new Transaction();
+    transaction.setAccount(account);
+    transaction.setFee(transactionFee);
+    transaction.setSequence(sequence);
+    transaction.setPayment(payment);
+    transaction.setSigningPublicKeyHex(publicKey);
+
+    // WHEN the transaction is serialized to JSON.
+    const serialized = Serializer.transactionToJSON(transaction);
+
+    // THEN the result is as expected.
+    const expectedJSON = {
+      Account: account,
+      Amount: value + "",
```

Comment:
NIT: casting value to a string with the + "" is the fastest way to convert an int to a string but I'm not a fan of it from syntax wise

Reply:
Assume `value.toString()` works, LMK if you prefer something else. 

Fixed here and elsewhere.

Reply:
Resolving since Tyler approved changes.